client/src/views/Authentication/AuthLoginComponent.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+    <div :class="loginClassList">
+
+      <q-input v-model.lazy="email" label="Email" :dense="dense" :rules="[val => checkEmailField(val)]"/>
+
+      <q-input v-model.lazy="password" :type="isPwd1 ? 'password' : 'text'" label="Password" :dense="dense" :rules="[val => checkPasswordField(val)]">
+        <template v-slot:append>
+          <q-icon
+            :name="isPwd1 ? 'visibility_off' : 'visibility'"
+            class='cursor-pointer'
+            @click='isPwd1 = !isPwd1'
+          />
+        </template>
+      </q-input>
+
+      <p id='error-message'>{{ error }}</p>
+
+      <q-btn color='white' text-color='black' @click='login' class='full-width' :disabled='!enableLogin'>log in
+        <q-spinner-bars
+          class='q-ml-md'
+          color='primary'
+          size='1em'
+          v-show='loading'
+        />
+      </q-btn>
+
+      <p class='paragraph'><a href='/forgotpassword'>Forgot password?</a></p>
+
+      <p id='center-paragraph'>OR</p>
+
+      <q-btn color='white' text-color='black' @click="handleClickSignIn" class='full-width'>
+        <img id='google-img' src='../../assets/google-icon.png' />
+        Sign In
+      </q-btn>
+
+      <p class='paragraph'>Don't have account yet? <a href='/register'>Register</a></p>
+
+    </div>
+</template>
+
+<script>
+import axios from 'axios'
+import { validationMixin } from '../../utilities/validationMixin.js'
+import { notificationMixin } from '../../utilities/notificationMixin.js'
+
+export default {
+  mixins: [validationMixin, notificationMixin],
+  props: ['loginClassList'],
+  data () {
+    return {
+      email: '',
+      password: '',
+      dense: false,
+      isPwd1: true,
+      loading: false,
+      emailField: false,
+      passwordField: false,
+      error: ''
+    }
+  },
+  methods: {
+    login () {
+      this.loading = true
+      const { email, password } = this
+      axios
+        .post('api/auth/login', { password, email })
+        .then(res => {
+          this.$store.commit('login', { token: res.data.token, user: res.data.token })
+          this.setAxiosHeaders(res.data.token)
+          this.loading = false
+          this.$router.push('/feed')
+        })
+        .catch(err => {
+          if (err) {
+            this.error = err.response.data.error
+          } else {
+            this.error = 'Sorry, something went wrong...'
+          }
+          this.loading = false
+        })
+    },
+    setAxiosHeaders (token) {
+      axios.defaults.headers.common['Authorization'] = 'Bearer ' + token
```

Comment:
Please don't keep such logic inside a component, you should move to some sort of ApiService, etc. where you can set default headers. And then import setApiHeaders or setApiAuthorizationHeaders it's up to you.